src/librustc_trans/back/linker.rs

```diff
@@ -456,7 +457,8 @@ fn exported_symbols(scx: &SharedCrateContext,
 
     // See explanation in GnuLinker::export_symbols, for
     // why we don't ever need dylib symbols on non-MSVC.
-    if crate_type == CrateType::CrateTypeDylib {
+    if crate_type == CrateType::CrateTypeDylib ||
+       crate_type == CrateType::CrateTypeRustcMacro {
```

Comment:
Ah this is actually kinda subtle b/c cydlib is also dynamically linked. I do agree though that the number of places that had to be updated with a new crate type was unfotunate :(